backend/canisters/group/impl/src/updates/delete_messages.rs

```diff
@@ -77,14 +86,17 @@ fn delete_messages_impl(args: Args, runtime_state: &mut RuntimeState) -> Respons
         let remove_deleted_message_content_at = now + (5 * MINUTE_IN_MS);
         for (message_id, result) in delete_message_results {
             if matches!(result, DeleteMessageResult::Success) {
-                runtime_state.data.timer_jobs.enqueue_job(
-                    TimerJob::RemoveDeletedMessageContent(RemoveDeletedMessageContentJob {
-                        thread_root_message_index: args.thread_root_message_index,
-                        message_id,
-                    }),
-                    remove_deleted_message_content_at,
-                    now,
-                );
+                // After 5 minutes hard delete those messages where the deleter was the message sender
+                if my_messages.contains(&message_id) {
```

Comment:
Maybe move this filter to be on the iterator on line 78.
We could move the match statement too.